### PR TITLE
Display message when there are no related items.

### LINKED ISSUE #31: Product detail page does not load
**Is your feature request related to a problem? Please describe.**
When clicking on a product to view its details page, there is an error. The related products section is creating this error. 

**Describe the solution you'd like**
For now, add conditional rendering so that the page will still load even when the related products section is broken.

**Additional context**
We will want to remove/refactor this once we find the actual bug, see issue #27

### PR DESCRIPTION
**Pull request by elsafluss**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [ ] Adds a new feature
- [ ] Improves an existing feature
- [x] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
When clicking on an item to view its detail page, an error occured.
This adds a temporary ternary so that we can get to the page and check for any other bugs.

Also updated lifecycle names.

## *What are the relevant tickets?*
Closes #31

## *Bugs:*
We still need to figure out why the list of related products is not showing up. Perhaps the action/dispatcher is returning the wrong thing?
On first render, `Related.js` has an empty array for `this.props.productsRelated.list` and that's fine, but then on a subsequent render it changes to `null` which is wrong.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: code/web/src/modules/product/Detail.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ class Detail extends PureComponent {
   }
 
   render() {
+    // console.log(this.state)
```
Comment (alia-peterson) [inline code/web/src/modules/product/Detail.js:45]: I thought you deleted this console.log before pushing but maybe I misunderstood. Want to delete before merging?

**Review by alia-peterson — CHANGES_REQUESTED** [review]
- [ ] Changes generate no new warnings or errors
- [ ] Branch was tested locally without failures
- [ ] All tests are passing as expected

File: code/web/src/modules/product/Detail.js (hunk 1 above)
Comment (elsafluss) [reply code/web/src/modules/product/Detail.js]: Good catch, I missed that one. I've removed it.

**Review by alia-peterson — APPROVED** [review]

File: code/web/src/modules/product/Related.js [hunk 2]
```diff
@@ -31,26 +30,29 @@ class Related extends PureComponent {
   }
 
   render() {
+
     const { isLoading, list } = this.props.productsRelated
 
     return (
       <div>
         {/* Related product list */}
+        {list ?
```
Comment (mattdeann) [inline code/web/src/modules/product/Related.js:39]: We should come back to this later, move the logic out of return

**Review by mattdeann — APPROVED** [review]
- [x] Changes generate no new warnings or errors
- [ ] Branch was tested locally without failures
- [ ] All tests are passing as expected

**Pull request merged**